applications/FluidDynamicsApplication/python_scripts/navier_stokes_solver_vmsmonolithic.py

```diff
@@ -189,9 +190,14 @@ def _ValidateSettings(self, settings):
             "turbulence_model": "None"
         }""")
 
-        settings = self._BackwardsCompatibilityHelper(settings)
-        settings.ValidateAndAssignDefaults(default_settings)
-        return settings
+        default_settings.AddMissingParameters(super(NavierStokesSolverMonolithic, cls).GetDefaultSettings())
```

Comment:
```suggestion
        default_settings.AddMissingParameters(super(NavierStokesSolverMonolithic, cls).self._BackwardsCompatibilityHelper(GetDefaultSettings()))
```

would this work @marcnunezc ?

Reply:
I would say no, here you would input the default settings to the _BackwardsCompatibilityHelper, but the settings that should enter there are the user-defined settings.

Reply:
ok now I think I understood
TBH I didn't think this far, I only ported StructuralMechanics at the time => see [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py#L38-L57)
=> I think you could do the same, i.e. call the `_BackwardsCompatibilityHelper` in the constructor before you pass the settings on to the baseclass

Reply:
I implemented your suggestion to avoid overriding the base method.